### PR TITLE
PostgreSQL: added support for `CREATE ROLE` and `DROP ROLE` statements

### LINKED ISSUE #2943: Found unparsable section errors for postgres dialect
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I'm running `sqlfluff lint .` against the next file I have the next errors:
```
DROP ROLE IF EXISTS supervisor;
DROP ROLE IF EXISTS nurse;
DROP ROLE IF EXISTS family_member;
CREATE ROLE supervisor;
CREATE ROLE nurse;
CREATE ROLE family_member;
```

```
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'DROP ROLE                                     
                       | IF EXISTS supervisor'
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'DROP ROLE
                       | IF EXISTS nurse'
L:   5 | P:   1 |  PRS | Line 5, Position 1: Found unparsable section: 'DROP ROLE
                       | IF EXISTS family_member'
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'CREATE
                       | ROLE supervisor'
L:   7 | P:   1 |  PRS | Line 7, Position 1: Found unparsable section: 'CREATE
                       | ROLE nurse'
L:   8 | P:   1 |  PRS | Line 8, Position 1: Found unparsable section: 'CREATE
                       | ROLE family_member'
```

What can be the reason? 

### Expected Behaviour

No errors are found

### Observed Behaviour

Errors found
```
DROP ROLE IF EXISTS supervisor;
DROP ROLE IF EXISTS nurse;
DROP ROLE IF EXISTS family_member;
CREATE ROLE supervisor;
CREATE ROLE nurse;
CREATE ROLE family_member;
```

```
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'DROP ROLE                                     
                       | IF EXISTS supervisor'
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'CREATE
                       | ROLE supervisor'

```

### How to reproduce

run sqlfluff against the next sql:
```
DROP ROLE IF EXISTS supervisor;
DROP ROLE IF EXISTS nurse;
DROP ROLE IF EXISTS family_member;
CREATE ROLE supervisor;
CREATE ROLE nurse;
CREATE ROLE family_member;
```

```
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'DROP ROLE                                     
                       | IF EXISTS supervisor'
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'CREATE
                       | ROLE supervisor'

```

### Dialect

postgres

### Version

0.11.2

### Configuration

[sqlfluff]
dialect = postgres
exclude_rules = L057

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by dnim**
### Brief summary of the change made
fixes #2943

### Are there any other side effects of this change that we should be aware of?
everything looks straightforward, so shouldn't be

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -3278,7 +3278,9 @@ class CreateTypeStatementSegment(BaseSegment):
         Bracketed(Delimited(Anything()), optional=True),
     )
 
-
+############################
+# ROLE
+############################
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_ansi.py:3283]: We let the doc string do the talking so don't need this:

```suggestion

```

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 2]
```diff
@@ -3293,6 +3295,26 @@ class CreateRoleStatementSegment(BaseSegment):
...
+    https://docs.exasol.com/sql/drop_role.htm
+    """
+
+    type = "drop_role_statement"
+
+    is_ddl = False
+    is_dml = False
+    is_dql = False
+    is_dcl = True
+
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_ansi.py:3310]: Only Exasol uses these (and not entirely sure they should! But that's another question).

```suggestion
```

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 3]
```diff
@@ -3293,6 +3292,21 @@ class CreateRoleStatementSegment(BaseSegment):
         Ref("ObjectReferenceSegment"),
     )
 
+class DropRoleStatementSegment(BaseSegment):
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_ansi.py:3295]:
```suggestion

class DropRoleStatementSegment(BaseSegment):
```

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 4]
```diff
@@ -3278,7 +3278,6 @@ class CreateTypeStatementSegment(BaseSegment):
         Bracketed(Delimited(Anything()), optional=True),
     )
 
-
 class CreateRoleStatementSegment(BaseSegment):
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_ansi.py:3281]:
```suggestion

class CreateRoleStatementSegment(BaseSegment):
```

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 5]
```diff
@@ -3293,6 +3295,26 @@ class CreateRoleStatementSegment(BaseSegment):
         Ref("ObjectReferenceSegment"),
     )
 
+class DropRoleStatementSegment(BaseSegment):
+    """A `DROP ROLE` statement with CASCADE option.
+
+    https://docs.exasol.com/sql/drop_role.htm
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_ansi.py]: Updated this. If you have the coachroach link then let's add it.

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 6]
```diff
@@ -1079,21 +1079,25 @@ class WithinGroupClauseSegment(BaseSegment):
     )
 
 
-class CreateRoleStatementSegment(ansi.CreateRoleStatementSegment):
+class CreateRoleStatementSegment(BaseSegment):
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_postgres.py]: I've updated this.

**Review by tunetheweb — APPROVED** [review]
I think this is good to go now!

**Comment by dnim** [comment]
@tunetheweb Thank you for your support and explanation! Next time will be more accurate :)

**Comment by tunetheweb** [comment]
No probs. And sorry for just barging and taking over, rather than letting you resolve them, but looking to release 0.12.0 soon and wanted to get this included in that.

**Pull request merged**